### ISSUE TITLE
Ability to tell checkout to clone the parent project recursively instead of a subproject.

### ISSUE THREAD

**Issue by AraHaan** [actions/checkout#431]
This is a feature request for checkout, basically I want to tell checkout to skip checking out the current repository (if I know it is a submodule that depends on files from the parent repository) in order to build properly on that project.

One could say, why not just use a single repository for the projects, Well sometimes it's impractical or impossible because of a number of reasons including the fact that I am waiting months for someone to pull request an performance optimization for that project and so must wait until then to make it all a single repository.

Basically here is what I mean, I will show you the link to the subproject: https://github.com/Elskom/zlib.managed
What I want is for instead on actions where it uses this action to be able to clone instead the parent https://github.com/Elskom/Sdk recursively which will clone that repository too but ensures all files it depends on are cloned as well so that way I would know accurately if someone's changes really did break build or not.

**Comment by AraHaan** [comment]
Now that I think of it I would not be able to recurisively checkout the submodule using the parent project, but checkout the parent then checkout the submodule manually using the ref that invoked the action on the submodule.
